File: models/parking_slot.py
class ParkingSlot:
    def __init__(self, slot_id: str, location: str, vehicle_type: str):
        self.slot_id = slot_id
        self.location = location
        self.vehicle_type = vehicle_type
        self.is_available = True

    def occupy(self):
        self.is_available = False

    def release(self):
        self.is_available = True

File: backend/models/ParkingSlot.js
const mongoose = require("mongoose");
//import mongoose from "mongoose";

const parkingSlotSchema = new mongoose.Schema({
  slotId: { type: String, required: true },
  location: String,
  vehicleType: String,
  isAvailable: { type: Boolean, default: true }
});

//export default mongoose.model("ParkingSlot", parkingSlotSchema);
module.exports = mongoose.model("ParkingSlot", parkingSlotSchema);
